scheduler/generic_sched.go

```diff
@@ -512,7 +512,13 @@ func (s *GenericScheduler) findPreferredNode(allocTuple *allocTuple) (node *stru
 			return
 		}
 		if taskGroup.LocalDisk.Sticky == true {
-			node, err = s.state.NodeByID(allocTuple.Alloc.NodeID)
+			var preferredNode *structs.Node
```

Comment:
@dadgar System sched always creates a missing Alloc Tuple with a particular node id. So we don't need this in the system scheduler. Nodes which are in drain mode don't get any new allocations.